Xamarin.Forms.Xaml/MarkupExtensions/OnIdiomExtension.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+
+namespace Xamarin.Forms.Xaml
+{
+	public class OnIdiomExtension : IMarkupExtension
+	{
+		// See Device.Idiom
+
+		public object Default { get; set; }
```

Comment:
could just be string in your case...

as the properties aren't typed, you won't get intelissense completion unless special-cased

Reply:
By making these `object`, we can allow further markup extensions to set those values (i.e. `StaticResource`?)

Reply:
@kzu when I wrote the original change the idea was to make everything an object so we could use this on more than just strings. Since this is a new platform markup extension we should support more than just strings with it. This will open up a lot of flexibility for developers to have let's say different images or objects inserted into their XAML based on the platform or idiom which is a powerful thing.